### DISCUSSION TITLE
Add a method for possible transtion

### DISCUSSION THREAD

**Discussion by AndyOGo** [statelyai/xstate#1864] [Ideas]
Instead of having to remember the name of the event, as in `send('SOME_EVENT');`.
It would be easier to have a corresponding method automatically generated, like `someEvent()`.

**Comment by davidkpiano** [comment]
This is what TypeScript is useful for. Generating methods is too "magical" - you're supposed to be explicit.

You can do this in userland, however. The goal for XState isn't to make things "easier", it's to make things more explicit. It's definitely possible to make things easier, but XState needs to remain unopinionated about that - you can do it yourself.

**Reply by AndyOGo** [reply]
Thanks for your quick reply.

**Reply by tomByrer** [reply]
Could also call all the happy-path events "NEXT", or get the `nextEvents` array from the returned state object. to make names for the coming `send()` event...
I plan to do one of these to make my event button rendering automated.
